be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -127,6 +127,8 @@ StatusOr<RowsetSharedPtr> BetaRowsetWriter::build() {
         // if load only has delete, we can skip the partial update logic
         if (_context.partial_update_tablet_schema && _flush_chunk_state != FlushChunkState::DELETE) {
             DCHECK(_context.referenced_column_ids.size() == _context.partial_update_tablet_schema->columns().size());
+            RETURN_IF(_num_segment != _rowset_txn_meta_pb->partial_rowset_footers().size(),
```

Comment:
Add concise commit message to describe the bug and solution

Reply:
added.